orb/core/collection.py

```diff
@@ -97,8 +105,35 @@ def __len__(self):
         return self.count()
 
     def __iter__(self):
-        for record in self.records():
-            yield record
+        if self.isNull():
```

Comment:
what this was doing before was creating a list in the `self.records()` call and then looping through that list and yielding the values.  moved the general logic per record to the `_process` method, and am able to yield individual results now